lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart

```diff
@@ -818,6 +825,37 @@ class MailboxDashBoardController extends ReloadableController
     }
   }
 
+  void injectAIScribeBindings(Session? session, AccountId? accountId) {
+    try {
+      if (session == null || accountId == null) return;
+
+      final aiCapability = session.getAICapability(accountId);
+      final scribeEndpoint = aiCapability?.scribeEndpoint;
+
+      // Delete existing AI bindings if they exist
+      if (Get.isRegistered<AIDataSourceImpl>()) {
+        Get.delete<AIDataSourceImpl>(force: true);
+      }
+      if (Get.isRegistered<AIDataSource>()) {
+        Get.delete<AIDataSource>(force: true);
+      }
+      if (Get.isRegistered<AIScribeRepositoryImpl>()) {
+        Get.delete<AIScribeRepositoryImpl>(force: true);
+      }
+      if (Get.isRegistered<AIScribeRepository>()) {
+        Get.delete<AIScribeRepository>(force: true);
+      }
+      if (Get.isRegistered<GenerateAITextInteractor>()) {
+        Get.delete<GenerateAITextInteractor>(force: true);
+      }
+
+      // Reinitialize with the correct endpoint
+      AIScribeBindings(scribeEndpoint: scribeEndpoint).dependencies();
```

Comment:
how about to change the approach, instead of check and delete then re-initialize why not using `DioInterceptor` to setup endpoint for dio, then we don't care about `isRegistered`?